mitodl/open-discussions
# Added video favoriting functionality

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2332 

#### What's this PR do?
- Adds frontend and backend code to support favorites for videos
- Refactored favorite/unfavorite tests for courses to parametrize on the factory and route (added a case for videos and missing ones for bootcamps and programs while I was in there)

#### How should this be manually tested?
Follow testing instructions in #2327 to get some test videos indexed

Toggle the favorite button on the course search page, you should see the UI update and you should also see it come/go from `/api/v0/favorites`
